libs/sdk-core/src/breez_services.rs

```diff
@@ -879,11 +879,19 @@ impl BreezServices {
         self.connect_lsp_peer(node_pubkey).await?;
 
         // First query the changes since last sync time.
-        let since_timestamp = self.persister.last_payment_timestamp().unwrap_or(0);
+        let since_timestamp = self
+            .persister
+            .get_last_sync_time()?
+            .or_else(|| self.persister.last_payment_timestamp().ok())
+            .unwrap_or(0);
         let new_data = &self
             .node_api
             .pull_changed(since_timestamp, balance_changed)
             .await?;
+        let last_sync_time = SystemTime::now()
```

Comment:
I think it is better to use the last_sync_time as the newest payment, received or sent (but not pending), time.
I think the time should be taken from gl rather the local time because:
1. It avoids any clock sync between local time and gl (such as if a payment would be later added exactly at that gap).
2. Eventually we want to have an endpoint in gl that filters based on timestamp.

Reply:
👍  these were my concerns too